packages/zksend/src/experimental/wallet/index.ts

```diff
@@ -0,0 +1,468 @@
+				JSON.stringify({
+					accounts: filteredAccounts,
+					token,
+				}),
+			);
+		}
+
+		this.#accounts = this.#accounts.filter((account) => account.address !== address);
+
+		if (this.#accounts.length === 0) {
+			this.#handleWalletIframeDisconnect();
+		}
+
+		this.#events.emit('change', { accounts: this.accounts });
+	}
+
+	#setAccounts(accounts?: StashedAccount[]) {
+		if (accounts && accounts.length) {
+			this.#accounts = accounts.map((account) => {
+				return new ReadonlyWalletAccount({
+					address: account.address,
+					chains: [SUI_MAINNET_CHAIN],
+					features: ['sui:signTransactionBlock', 'sui:signPersonalMessage'],
+					publicKey: account.publicKey ? fromBase64(account.publicKey) : new Uint8Array(),
+				});
+			});
+		} else {
+			this.#accounts = [];
+		}
+
+		this.#events.emit('change', { accounts: this.accounts });
+	}
+
+	#connect: StandardConnectMethod = async (input) => {
+		if (input?.silent) {
+			const { accounts } = getStashedSession();
+
+			if (accounts.length) {
+				this.#setAccounts(accounts);
+			}
+
+			this.#embedStashedIframe();
+
+			return { accounts: this.accounts };
+		}
+		const popup = new StashedPopup({
+			name: this.#name,
+			origin: this.#origin,
+		});
+
+		const response = await popup.send({
+			type: 'connect',
+		});
+
+		if (!('accounts' in response)) {
+			throw new Error('Unexpected response');
+		}
+
+		localStorage.setItem(
+			STASHED_SESSION_KEY,
+			JSON.stringify({ accounts: response.accounts, token: response.session }),
+		);
+
+		this.#setAccounts(response.accounts);
+
+		this.#embedStashedIframe();
+
+		return { accounts: this.accounts };
+	};
+
+	#handleWalletIframeDisconnect = () => {
+		this.#walletStatusCheckEnabled = false;
+
+		setTimeout(() => {
+			if (this.#embeddedIframe) {
+				document.body.removeChild(this.#embeddedIframe);
+			}
+			window.removeEventListener('message', () => {});
+			this.#clearWalletStatusInterval();
+		}, 2000);
+	};
+
+	#disconnect: StandardDisconnectMethod = async () => {
+		this.#embeddedIframe?.contentWindow?.postMessage(
+			{
+				type: 'WALLET_DISCONNECTED',
+				payload: getPostMessagePayload(),
+				session: getStashedSession().token,
+			},
+			this.#origin,
+		);
+
+		localStorage.removeItem(STASHED_SESSION_KEY);
+
+		this.#setAccounts();
+
+		this.#handleWalletIframeDisconnect();
+	};
+}
+
+export function registerStashedWallet(
+	name: string,
+	{
+		origin,
+	}: {
+		origin?: string;
+	} = {},
+) {
+	const wallets = getWallets();
+
+	let addressFromRedirect: string | null = null;
+
+	try {
+		const params = new URLSearchParams(window.location.search);
+		addressFromRedirect = params.get('stashed_address') || params.get('zksend_address');
+	} catch {
+		// Ignore errors
+	}
+
+	const stashedWalletInstance = new StashedWallet({
+		name,
+		origin,
+	});
+
+	stashedWalletOrigin = origin || DEFAULT_STASHED_ORIGIN;
+
+	const unregister = wallets.register(stashedWalletInstance);
+
+	/* @ts-ignore */
+	if (window.stashed) {
```

Comment:
Instead of doing this, use the wallet standard to:
- Get the list of wallets
- Subscribe to the list of wallets

If the Stashed wallet ever appears, then un-register it.